File: Espd/typings/Evidence.ts
import { IdentifierType } from './IdentifierType'
import { CodeType } from './CodeType'
import { Party } from './Party'
import { DocumentReference } from './DocumentReference'
import { Language } from './Language'
import assign from 'lodash.assign'

interface RequiredArgsEvidence {
}

export class Evidence {
  public constructor(init?:Partial<Evidence> & RequiredArgsEvidence) {
    assign(this, init)
  }
  public id?: IdentifierType
  public evidenceTypeCode?: CodeType
  public name?: string
  public description?: string[]
  public candidateStatement?: string[]
  public confidentialityLevelCode?: CodeType
  public evidenceIssuingParty?: Party
  public documentReference?: DocumentReference[]
  public language?: Language
  
}

File: UBL/CommonAggregateComponents/Evidence.cs
using System;
using System.Linq;
using System.Xml.Linq;
using Hilma.UBL.Attributes;
using Hilma.UBL.Serializers;
using Hilma.UBL.UnqualifiedDataTypes;

namespace Hilma.UBL.CommonAggregateComponents
{
  [Contract]
    public class Evidence
    {
        public IdentifierType ID { get; set; }
        public CodeType EvidenceTypeCode { get; set; }
        public string Name { get; set; }
        public string[] Description { get; set; }
        public string[] CandidateStatement { get; set; }
        public CodeType ConfidentialityLevelCode { get; set; }
        public Party EvidenceIssuingParty { get; set; }
        public DocumentReference[] DocumentReference { get; set; }
        public Language Language { get; set; }

        public XElement Serialize(string name = null)
        {
          return new XElement(UblNames.Cac + ( name ?? nameof(Evidence)),
            ID.Serialize(nameof(ID)),
            EvidenceTypeCode.Serialize(nameof(EvidenceTypeCode)),
            Name.Serialize(nameof(Name)),
            Description.Serialize(nameof(Description)),
            CandidateStatement.Serialize(nameof(CandidateStatement)),
            ConfidentialityLevelCode.Serialize(nameof(ConfidentialityLevelCode)),
            EvidenceIssuingParty?.Serialize(nameof(EvidenceIssuingParty)),
            DocumentReference?.Select( dr => dr.Serialize())
            );
        }
    }

}